Addressing some test gaps for MethodCallExpression

bartdesmet opened the pull request:
Some cases dealing with instance/static method calls with 0 arguments, and with N arguments where N is larger than the maximum argument count for which we provide optimized layouts.

stephentoub commented on src/System.Linq.Expressions/tests/Call/CallFactoryTests.cs:
```diff
@@ -154,12 +254,32 @@ private static void AssertCallIsOptimized(MethodCallExpression expr, Expression
-            var updated = Update(expr);
-            var visited = Visit(expr);
+            var updatedArgs = UpdateArgs(expr);
+            var visitedArgs = VisitArgs(expr);
```
Can you use the real type instead of var for these two?

stephentoub commented on src/System.Linq.Expressions/tests/Call/CallFactoryTests.cs:
```diff
@@ -154,12 +254,32 @@ private static void AssertCallIsOptimized(MethodCallExpression expr, Expression
+            }
 
-            foreach (var node in new[] { expr, updated, visited })
+            foreach (var node in nodes)
```
Nit: var => ?

bartdesmet commented:
Thanks! I'll run my Roslyn code fix on the test code to address var in bulk. It was only run on the product code before.

bartdesmet commented:
Addressed code review feedback. There will be merge conflicts with my bulk replacement of unwarranted `var` usage, but I'll deal with that. Any order to merge these is fine :-).
